src/sagemaker/predictor.py

```diff
@@ -105,9 +105,25 @@ def _create_request_args(self, data, initial_args=None):
         args['Body'] = data
         return args
 
-    def delete_endpoint(self):
-        """Delete the Amazon SageMaker endpoint backing this predictor.
+    def _delete_endpoint_config(self):
+        """Delete the Amazon SageMaker endpoint configuration
         """
+        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
+        endpoint_config_name = endpoint_description['EndpointConfigName']
+        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+
+    def delete_endpoint(self, delete_endpoint_config=True):
+        """Delete the Amazon SageMaker endpoint backing this predictor. Also delete the endpoint configuration attached
+           to it if delete_endpoint_config is True.
+
+        Args:
+            delete_endpoint_config (bool): Flag to indicate whether to delete endpoint configuration together with
```

Comment:
Drop-in

"Flag to indicate whether or not to delete the corresponding SageMaker endpoint configuration tied to the endpoint. If False, only the endpoint will be deleted. (default: True)"

Nitpick: Please wrap the default with parentheses. Similar to https://github.com/aws/sagemaker-python-sdk/blob/757c8ab8a48000344e37bc0683348e9060ffde17/src/sagemaker/predictor.py#L51